x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/date_range_picker.tsx

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useDispatch } from 'react-redux';
+import React, { memo, useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import moment, { Moment } from 'moment';
+import { EuiFlexGroup, EuiFlexItem, EuiDatePicker, EuiDatePickerRange } from '@elastic/eui';
+
+import * as i18 from '../../translations';
+import { useEndpointSelector } from '../../hooks';
+import { getActivityLogDataPaging, selectedAgent } from '../../../store/selectors';
+
+const DatePickerWrapper = styled.div`
+  width: ${(props) => props.theme.eui.fractions.single.percentage};
+  background: white;
+`;
+const StickyFlexItem = styled(EuiFlexItem)`
+  position: sticky;
+  top: ${(props) => props.theme.eui.euiSizeM};
+  z-index: 1;
+`;
+
+export const DateRangePicker = memo(() => {
+  // clear dates on endpoint selection
+  const elasticAgentId = useEndpointSelector(selectedAgent);
+  useEffect(() => {
+    return () => {
```

Comment:
I don't think you need to clear this when component unmount as this is a local state. Is there some specific reason you are doing this?

Reply:
It's to clear the date range values when you select another endpoint from the endpoint list.

Reply:
The component is not unmounted/mounted again?

Reply:
It is re-rendered yes. But since the dates are being set from the redux store they don't change on a re-render. The dates change only when the page update action is dispatched. But your suggestion about using the values from the store itself will fix the problem. See below. 👇

Reply:
done f551b67d661a330621fb77c435db26ec90288b81